packages/gatsby/src/utils/page-data-helpers.ts

```diff
@@ -0,0 +1,50 @@
+import { IGatsbyPage } from "../redux/types"
+
+export interface IPageData {
+  componentChunkName: IGatsbyPage["componentChunkName"]
+  matchPath?: IGatsbyPage["matchPath"]
+  path: IGatsbyPage["path"]
+  staticQueryHashes: Array<string>
+}
+
+export function constructPageDataString(
+  {
+    componentChunkName,
+    matchPath,
+    path: pagePath,
+    staticQueryHashes,
+  }: IPageData,
+  result: string | Buffer
+): string {
+  let body = `{
+    "componentChunkName": "${componentChunkName}",
+    "path": "${pagePath}",
```

Comment:
I think we can safely change it along the way to fix https://github.com/gatsbyjs/gatsby/issues/32892. But not insisting - could be a separate PR as well.
```suggestion
    "path": JSON.stringify(pagePath),
```

Reply:
Makes sense, comitted that suggestion in